application/helpers/common_helper.php

```diff
@@ -943,17 +942,28 @@ function returnGlobal($stringname, $bRestrictToString = false)
 }
 
 
-function sendCacheHeaders()
+function sendSurveyHttpHeaders()
 {
     if (!headers_sent()) {
+        // Default headers für surveys
+        $headers = [
+                     'Cache-Control: no-store, no-cache, must-revalidate, max-age=0, post-check=0, pre-check=0',
+                     'Pragma: no-cache',
+                     'Content-Type: text/html; charset=utf-8'
+                    ];
         if (Yii::app()->getConfig('x_frame_options', 'allow') == 'sameorigin') {
-            header('X-Frame-Options: SAMEORIGIN');
+            $headers[] = 'X-Frame-Options: SAMEORIGIN';
+        }
+        // plugins can modify the
+        $event = new PluginEvent('beforeSurveyHttpHeaders');
+        $event->set('headers', $headers);
+        App()->getPluginManager()->dispatchEvent($event);
+        $headers = $event->get('headers', []);
```

Comment:
An idea to allow easier replacement , unsure it's good but i think it can be great.

```
       $headers = [       
            'Cache-Control' => 'Cache-Control: no-store, no-cache, must-revalidate, max-age=0, post-check=0, pre-check=0',
            'Pragma' => 'Pragma: no-cache',
            'Content-Type' => 'Content-Type: text/html; charset=utf-8'
        ];
        if (Yii::app()->getConfig('x_frame_options', 'allow') == 'sameorigin') {
            $headers['X-Frame-Options'] = 'X-Frame-Options: SAMEORIGIN';
        }
```

Then plugins can replace specific core header (or gentle plugin header added too).

Reply:
Arg too late ;)